src/NewTools-Debugger/Halt.extension.st

```diff
@@ -0,0 +1,8 @@
+Extension { #name : #Halt }
+
+{ #category : #'*NewTools-Debugger' }
+Halt >> description [
+	(self signalContext receiver isKindOf: Exception) 
+		& (self signalContext selector = #signal) ifTrue: [ ^ 'Halt' ].
```

Comment:
I think that this is bad. `#isKindOf:` is costly and the behavior can be reproduced by calling `StDebuggerContextPredicate>>#printDescriptionPrefixOn:`. Using this method would also allow to handle post-mortems (which are just dead contexts: contexts that have already terminated in which we can't step anymore).

Overriding `#description` in the class `Halt` causes the debugger to display `Halt` when a breakpoint is hit, because the class ` Break` inherits from `Halt`.

From looking at the code of `StDebuggerActionModel>>#printHaltDescription:`, I would say that the fact that "Halt in" not being displayed by the debugger when executing `Halt now` is a bug, so trying to reproduce it like that is not good.